### ISSUE TITLE
Pass ClientHello by reference to ResolvesServerCert

### ISSUE THREAD

**Issue by blind-oracle** [rustls/rustls#1908]
**Checklist**
* [x] I've searched the issue tracker for similar requests

**Is your feature request related to a problem? Please describe.**

I'm doing some complex TLS certificate resolving logic and the fact that `ResolvesServerCert` trait takes `ClientHello` by value rather than by reference is a bit annoying. Since it also does not implement `Clone` and cannot be created (`new` is private to crate) I cannot pass it to more than one other implementer of `ResolvesServerCert` and that's sometimes needed.

I understand that it's minor issue and would break API, but maybe in some future versions it can be improved.

Thanks!

**Describe the solution you'd like**
Pass `ClientHello` by reference / make it `Clone` / wrap in `Arc`

**Comment by djc** [comment]
Have you looked at the [`Acceptor`](https://docs.rs/rustls/latest/rustls/server/struct.Acceptor.html) API? It is quite a bit more powerful and could probably handle your use case.

**Comment by blind-oracle** [comment]
@djc Yes, thanks, I know about it, but for me `ResolvesServerCert` is mostly enough. I solved my case by adding my own internal trait that takes `ClientHello` by ref, so at least for now it's not a problem. 

Just an idea for future releases that it might make people's life a bit easier.

Thanks!
